analyzers/src/SonarAnalyzer.VisualBasic/Rules/Hotspots/ExecutingSqlQueries.cs

```diff
@@ -44,34 +45,64 @@ public ExecutingSqlQueries()
-            context.Invocation is InvocationExpressionSyntax invocation
+            context.Node is InvocationExpressionSyntax invocation
                 ? invocation.ArgumentList.Get(index)
                 : null;
 
         protected override ExpressionSyntax GetArgumentAtIndex(ObjectCreationContext context, int index) =>
-            context.Expression is ObjectCreationExpressionSyntax objectCreation
+            context.Node is ObjectCreationExpressionSyntax objectCreation
                 ? objectCreation.ArgumentList.Get(index)
                 : null;
 
         protected override ExpressionSyntax GetSetValue(PropertyAccessContext context) =>
-            context.Expression is MemberAccessExpressionSyntax setter && setter.IsLeftSideOfAssignment()
+            context.Node is MemberAccessExpressionSyntax setter && setter.IsLeftSideOfAssignment()
                 ? ((AssignmentStatementSyntax)setter.GetSelfOrTopParenthesizedExpression().Parent).Right.RemoveParentheses()
                 : null;
 
-        protected override bool IsTracked(ExpressionSyntax argument, SemanticModel semanticModel) =>
-            argument != null
-            && (IsConcatenation(argument, semanticModel)
-                || argument.IsKind(SyntaxKind.InterpolatedStringExpression)
-                || (argument is InvocationExpressionSyntax invocation && IsInvocationOfInterest(invocation, semanticModel))
-                || IsTrackedVariableDeclaration(argument, semanticModel));
+        protected override bool IsTracked(ExpressionSyntax expression, SyntaxBaseContext context) =>
+            expression != null && (IsSensitiveExpression(expression, context.SemanticModel) || IsTrackedVariableDeclaration(expression, context));
+
+        protected override bool IsSensitiveExpression(ExpressionSyntax expression, SemanticModel semanticModel) =>
+            IsConcatenation(expression, semanticModel)
+            || expression.IsKind(SyntaxKind.InterpolatedStringExpression)
+            || (expression is InvocationExpressionSyntax invocation && IsInvocationOfInterest(invocation, semanticModel));
+
+        protected override Location SecondaryLocationForExpression(ExpressionSyntax node, string identifierNameToFind, out string identifierNameFound)
+        {
+            identifierNameFound = string.Empty;
+            if (node == null)
+            {
+                return Location.None;
+            }
+
+            if (node.Parent is EqualsValueSyntax equalsValue
+                && equalsValue.Parent is VariableDeclaratorSyntax declarationSyntax)
+            {
+                var identifier = declarationSyntax.Names.FirstOrDefault(name => name.Identifier.ValueText.Equals(identifierNameToFind, StringComparison.OrdinalIgnoreCase));
+                identifierNameFound = identifier != null ? identifier.Identifier.ValueText : string.Empty;
+                return identifier != null ? identifier.GetLocation() : Location.None;
```

Comment:
Line 84 assigns default value and that is re-assigned here for 2nd time. And the `if` condition is here twice. Let's rewrite it as
```
if(identifier==null)
{
    return Location.None;
}
else
{  
    identifierFound = identifier.Identifier.ValueText;
    return identifier.GetLocation();
}
```